src/fsharp/NameResolution.fs

```diff
@@ -1496,17 +1496,21 @@ type TcSymbolUseData =
 /// This is a memory-critical data structure - allocations of this data structure and its immediate contents
 /// is one of the highest memory long-lived data structures in typical uses of IDEs. Not many of these objects
 /// are allocated (one per file), but they are large because the allUsesOfAllSymbols array is large.
-type TcSymbolUses(g, capturedNameResolutions : ResizeArray<CapturedNameResolution>, formatSpecifierLocations: (range * int)[]) = 
-    
+type TcSymbolUses(g, capturedNameResolutions : ResizeArray<CapturedNameResolution>, formatSpecifierLocations: (range * int)[]) =
```

Comment:
I understand we chunk a ResizeArray, but if the ResizeArray is already on the LOH and we have to chunk it, should we do something about it as well?

Reply:
Yes, that is a problem still. @auduchinok suggested on the first revision that we should make some sort of LOH-aware ResizeArray that implemented `IList<'t>` by spanning across multiple chunked arrays, which would be a good solution to this. Then the `capturedNameResolutions` parameter could be that structure instead, and contain all the LOH-sizing concerns.

I would do this, but I have doubt in my ability to make that structure threadsafe. If threadsafety isn't a concern here I could have a go at writing that.

Reply:
That's fine, we can do this in another PR.

The ResizeArray is already not thread safe so don't worry about it; it's only accessed by the compiler service thread.